File: BE/src/main/java/com/ssafy/tax7i/banking/dto/AccountResponse.java
package com.ssafy.tax7i.banking.dto;

import com.ssafy.tax7i.banking.entity.Account;

import java.time.LocalDateTime;

public record AccountResponse(
        Long id,
        String accountType,       // enum.name() → "PERSONAL" | "BUSINESS"
        String bankCode,
        String accountNumber,
        String alias,
        Long balance,
        String status,            // enum.name() → "ACTIVE" | "INACTIVE" | "CLOSED"
        LocalDateTime createdAt
) {
    public static AccountResponse from(Account account, Long balance) {
        return new AccountResponse(
                account.getId(),
                account.getAccountType().name(),
                account.getBankCode(),
                account.getAccountNo(),
                account.getAlias(),
                balance,
                account.getStatus().name(),
                account.getCreatedAt()
        );
    }
}

File: FE/app/src/main/java/com/ssafy/seveniTax/data/model/pay/AccountResponse.kt
package com.ssafy.seveniTax.data.model.pay

data class AccountResponse(
    val id: Long,
    val accountType: String,
    val bankCode: String,
    val accountNumber: String,
    val alias: String?,
    val balance: Long,
    val status: String,
    val createdAt: String
)
